brave/app/scripts/controllers/bitgo.js

```diff
@@ -0,0 +1,193 @@
+const ObservableStore = require('obs-store')
+const bgUtil = require('brave-bitgo-client').util
+
+const supportedCoins = {
+  btc: 'Bitcoin',
+  bch: 'Bitcoin Cash',
+  btg: 'Bitcoin Gold',
+  bsv: 'Bitcoin SV',
+  zec: 'ZCash',
+  xlm: 'Stellar',
+  xrp: 'Ripple',
+  ltc: 'Litecoin',
+  dash: 'Dash',
+  algo: 'Algorand',
+  trx: 'Tron',
+  eos: 'Eos'
+}
+
+const uint8ToArrayBuf = (array) => {
+  return array.buffer.slice(array.byteOffset, array.byteLength + array.byteOffset)
+}
+
+const promiseGetSeed = (key) => {
+  return new Promise((resolve) => {
+    if (window.chrome && window.chrome.braveWallet) {
+      window.chrome.braveWallet.getBitGoSeed(key, resolve)
+    } else {
+      // polyfill
+      const seed = new Uint8Array(32)
+      window.crypto.getRandomValues(seed)
+      resolve(uint8ToArrayBuf(seed))
+    }
+  })
+}
+
+class BitGoController {
+  constructor (opts = {}) {
+    const initState = opts.initState || {}
+    this.proxyOrigin = opts.isProduction
+      ? 'https://bitgo-proxy.brave.com/'
+      : 'http://localhost:3000/'
+    this.store = new ObservableStore(initState)
```

Comment:
i think for this to work correctly between restarts, there needs to be something added to persist values in the store to disk and read it into initState when the controller is subsequently constructed

Reply:
unfortunately since the server can't tell which wallets were created by which users, i don't think the state for each crypto wallets user can be fetched from the server on init